pkg/apis/conditions.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apis
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+)
+
+// CopyConditions copies over explicitly marked conditions from one object to another
+func CopyConditions(ctx context.Context, from, to apis.ConditionsAccessor, allowed ...apis.ConditionType) {
+	fromConditions := from.GetConditions()
+	if len(fromConditions) == 0 {
+		return
+	}
+
+	conditions := make(apis.Conditions, 0, len(fromConditions))
+	for _, c := range fromConditions {
```

Comment:
N^2 :(
probably make one of them a map[] (or sets.String + cast) and check that way.

Reply:
Empirically our statuses don't have some massive number of condition types. So I'm not particularly concerned with theoretical runtime.

Thus I'd rather do the inner loop than do the extra memory allocation.

Reply:
resolving - this moved to `knative.dev/pkg`